ocaml/file_formats/cmi_format.mli

```diff
@@ -21,21 +21,26 @@ type pers_flags =
   | Opaque
   | Unsafe_string
 
-type cmi_infos = {
+type 'sg cmi_infos_generic = {
     cmi_name : Compilation_unit.t;
-    cmi_sign : Types.signature_item list;
+    cmi_sign : 'sg;
     cmi_crcs : Import_info.t array;
     cmi_flags : pers_flags list;
 }
 
+type cmi_infos_lazy = Subst.Lazy.signature cmi_infos_generic
+type cmi_infos = Types.signature cmi_infos_generic
+
 (* write the magic + the cmi information *)
-val output_cmi : string -> out_channel -> cmi_infos -> Digest.t
+val output_cmi : string -> out_channel -> cmi_infos_lazy -> Digest.t
 
 (* read the cmi information (the magic is supposed to have already been read) *)
 val input_cmi : in_channel -> cmi_infos
```

Comment:
Do we still need `input_cmi` and `read_cmi`? It seems to me that `input_cmi` is only used in a test and `read_cmi` in `extract_crc` - removing them would allow to simplify a bit the code of `cmi_format`

Reply:
These functions are used by things in our code base (both external and internal ones). It seemed much simpler to keep them than to update everything, including external libraries. In fact, I did remove them at first but then added them back in https://github.com/ocaml-flambda/flambda-backend/pull/1322/commits/b18e172f6cadf99aa42618f93f0c74b9d0fe56f6.